front/components/assistant_builder/instructions/InstructionsHistory.tsx

```diff
@@ -13,19 +13,43 @@ import { useCallback, useMemo } from "react";
 import React from "react";
 
 import { GaugeDiff } from "@app/components/assistant_builder/instructions/GaugeDiff";
-import type { LightAgentConfigurationType } from "@app/types";
+import { useEditors } from "@app/lib/swr/editors";
+import type {
+  LightAgentConfigurationType,
+  LightWorkspaceType,
+} from "@app/types";
 
 interface InstructionHistoryProps {
   history: LightAgentConfigurationType[];
   selectedConfig: LightAgentConfigurationType | null;
   onSelect: (config: LightAgentConfigurationType) => void;
+  owner: LightWorkspaceType;
+  agentConfigurationId: string | null;
+  currentInstructions: string;
 }
 
 export function InstructionHistory({
   history,
   onSelect,
   selectedConfig,
+  owner,
+  agentConfigurationId,
+  currentInstructions,
 }: InstructionHistoryProps) {
+  const { editors } = useEditors({
+    owner,
+    agentConfigurationId,
+    disabled: !agentConfigurationId,
+  });
```

Comment:
is it ok in terms of performances ?

Reply:
It should only be triggered when users open the dropdown